modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -1380,6 +1386,53 @@ impl Wallet {
         });
         (block_count_rx, fee_rate_rx)
     }
+
+    /// Shutdown the task group shared throughout fedimintd, giving 60 seconds
+    /// for other services to gracefully shutdown.
+    async fn graceful_shutdown(&self) {
+        if let Err(e) = self
+            .task_group
+            .clone()
+            .shutdown_join_all(Some(Duration::from_secs(60)))
+            .await
+        {
+            panic!("Error while shutting down fedimintd task group: {e}");
+        }
+    }
+
+    /// Returns once our bitcoin backend observes finality delay confirmations
+    /// of the consensus block count. If we don't observe enough confirmations
+    /// after one hour, we gracefully shutdown fedimintd. This is necessary
+    /// since we can no longer participate in consensus if our bitcoin backend
+    /// is unable to observe the same chain tip as our peers.
```

Comment:
Just a note: This shouldn't lead to significant latency spikes since we only get consensus on a block height after t/n guardians proposed it. So at worst these `t/n` guardians have higher latency than the typical contributors, but the ordering latency of the height votes makes it very likely that the remaining `f` nodes catch up to the chain tip in the meantime.